src/app/mod.rs

```diff
@@ -541,3 +543,50 @@ where
         Ok(())
     }
 }
+
+#[cfg(unix)]
+struct Ender {
+    sigint: Signal,
+    sigterm: Signal,
+}
+
+#[cfg(unix)]
+impl Ender {
+    fn new() -> Result<Self, std::io::Error> {
+        let sigint = signal(SignalKind::interrupt())?;
+        let sigterm = signal(SignalKind::terminate())?;
+
+        Ok(Ender { sigint, sigterm })
+    }
+    /// Waits for either an interrupt or terminate.  If interrupt is returned this equals `true`
+    async fn wait(&mut self) -> Result<bool, std::io::Error> {
```

Comment:
Using a `bool` here is not very intuitive. How about a custom `Enum` instead? Like

```rust
enum SigType {
    Interrupt,
    Terminate,
}
```